ci/release/apply_wheel_modifications.sh

```diff
@@ -19,5 +19,4 @@ sed -i "s/rmm/rmm${CUDA_SUFFIX}/g" python/pyproject.toml
 if [[ $CUDA_SUFFIX == "-cu12" ]]; then
     sed -i "s/cuda-python[<=>\.,0-9]*/cuda-python>=12.0,<13.0/g" python/pyproject.toml
     sed -i "s/cupy-cuda11x/cupy-cuda12x/g" python/pyproject.toml
-    sed -i "s/numba[<=>\.,0-9]*/numba>=0.57/g" python/pyproject.toml
```

Comment:
Just curious why is this no longer needed?

Reply:
Good question! This was needed to fix CUDA 12 wheels. Previously, we needed numba 0.57 (from PyPI) to make CUDA 12 work. However, 0.57 wasn't available for conda until today. To align these, we are requiring 0.57 everywhere and no longer need this workaround.